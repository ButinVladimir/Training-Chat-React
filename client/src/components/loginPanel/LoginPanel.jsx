import React from 'react';
import PropTypes from 'prop-types';
import { Socket } from 'socket.io-client';
import LoginForm from './loginForm/LoginFormContainer';
import LogoutForm from './logoutForm/LogoutFormContainer';
import './style.css';

function LoginPanel({
  loginned,
  socket,
}) {
  return (
    <div className="chat-login-panel-container">
      {!loginned && <LoginForm socket={socket} />}
      {loginned && <LogoutForm socket={socket} />}
    </div>
  );
}

LoginPanel.propTypes = {
  loginned: PropTypes.bool.isRequired,
  socket: PropTypes.instanceOf(Socket).isRequired,
};

export default LoginPanel;
