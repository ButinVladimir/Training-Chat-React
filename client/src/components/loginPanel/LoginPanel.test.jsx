import React from 'react';
import { Socket } from 'socket.io-client';
import { shallow } from 'enzyme';
import LoginPanel from './LoginPanel';

describe('LoginPanel', () => {
  let mockSocket;

  beforeEach(() => {
    mockSocket = new Socket();
  });

  it('renders without crashing when not loginned', () => {
    const component = shallow(
      <LoginPanel loginned={false} socket={mockSocket} />,
    );
    expect(component).toMatchSnapshot();
  });

  it('renders without crashing when loginned', () => {
    const component = shallow(
      <LoginPanel loginned socket={mockSocket} />,
    );
    expect(component).toMatchSnapshot();
  });
});
