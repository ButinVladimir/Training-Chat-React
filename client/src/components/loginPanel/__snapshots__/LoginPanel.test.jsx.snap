// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPanel renders without crashing when loginned 1`] = `
<div
  className="chat-login-panel-container"
>
  <Connect(LogoutForm)
    socket={lookup {}}
  />
</div>
`;

exports[`LoginPanel renders without crashing when not loginned 1`] = `
<div
  className="chat-login-panel-container"
>
  <Connect(LoginForm)
    socket={lookup {}}
  />
</div>
`;
